window/scene/mainScene/DragAndDrop.py

```diff
@@ -87,50 +96,21 @@ def dropEvent(self, event: QDropEvent):
     # ---------------------------------------------------------------------------- #
     #                                     Event                                    #
     # ---------------------------------------------------------------------------- #
-    def buttonClicked(self, url):
-        self.url = url
-        self.nextPage()
-    
-    def nextPage(self):
-        self.segmentTranscription = None
-
-        def setResultCallback(result):
-            self.segmentTranscription = result["segments"]
-
-
-        thread = threading.Thread(target=self.callback, args=(self.url, setResultCallback))
-        thread.start()
-        thread.join()
-        resultPage = UIResult(self.segmentTranscription, Utils.getDirectoryAbsolutePath() + "/export/video.mp4")
-        Application.getInstance().setActualPage(resultPage)
+    # This is the function called when we click on the button
+    def launchTranscription(self, url):
+        self.transcribe_thread = TranscribeVideoThread(url)
+        # Connect the callback to the finished signal !
+        self.transcribe_thread.finished_signal.connect(self.nextPage)
+        self.transcribe_thread.start()
+
+        # To avoid circular import here lol
+        from window.SceneManager import SceneManager
```

Comment:
i got a lot of problem with circular import so, i just decided to import SceneManager in the func, but it's probably not the best idea